src/averageAges.js

```diff
@@ -55,10 +75,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const filtredChildren = people.filter(child => {
+    return onlyWithSon
+      ? people.find(person =>
+        child.sex === PERSON_SEX_MALE && child.mother === person.name)
+      : people.some(person => child.mother === person.name);
```

Comment:
Try to simplify this 
<img width="1046" alt="image" src="https://github.com/mate-academy/js_average-ages/assets/86789161/bc589049-e8e1-4073-8f6c-24645d22c8ce">